pkg/forwarder/forwarder_health.go

```diff
@@ -144,14 +147,19 @@ func (fh *forwarderHealth) computeDomainsURL() {
 	}
 }
 
-func (fh *forwarderHealth) setAPIKeyStatus(apiKey string, domain string, status expvar.Var) {
+func (fh *forwarderHealth) setAPIKeyStatus(apiKey string, domain string, status *expvar.String) {
 	if len(apiKey) > 5 {
 		apiKey = apiKey[len(apiKey)-5:]
 	}
 	obfuscatedKey := fmt.Sprintf("API key ending with %s", apiKey)
-	apiKeyStatus.Set(obfuscatedKey, status)
+	if status == &apiKeyInvalid {
```

Comment:
I think you will want to remove the status from the other map in both cases here. Since (IIRC) the status can change at runtime.